internal/util/projutil/interactive_promt_util_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package projutil
+
+import (
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+// createInput mocks the stdin input to be provided to the
+// functions by creating a temp file and writing the data to
+// it.
+func createInput(content []byte) *os.File {
+	tempFile, err := ioutil.TempFile("", "test")
```

Comment:
NB: you could probably use a [pipe](https://golang.org/pkg/io/#Pipe) for this.

Reply:
In this function I was trying to mock `stdin` by writing data to a temp file as `os.stdin` is of type `*os.File`. pipe()  may not be completely helpful here. Even if data is written to pipe's writter end, it still cant be written to os.Stdin as stdin is not readable. In the sense, the following would not be possible either:
```
readp, writep := io.Pipe()
io.Copy(writep, data) // write data to pipe's writer

io.Copy(os.stdin, readp) // not possible as os.stdin is only writable not readable
```
creating a tempfile and using it as input on stdin seems to mock an entry of user input. A reader buffer is anyway being created inside the function which is being tested.

Reply:
For mocking external fd's like `os.Stdin` you could do something like:

```
func GetRequiredInput(msg string) string {
	return getRequiredInput(os.Stdin, msg)
}

func getRequiredInput(r io.Reader, msg string) string {
	for {
		fmt.Printf("%s", msg)
		value := readInput(r)
		if value != "" {
			return value
		}
		fmt.Printf("Input is required. ")
	}
}
```

So you aren't overwriting stdin's fd or creating files; then you could use a pipe, or more easily a `bytes.Buffer`.